fahrschein/src/main/java/org/zalando/fahrschein/NakadiReader.java

```diff
@@ -341,6 +342,7 @@ void runInternal() throws IOException, BackoffException {
                 throw e;
             }
         }
+        jsonInput.close();
```

Comment:
The closing logic is getting more and more complicated. Normally I would not recommend using exceptions for flow control, but maybe it would simpler to throw a custom exception, similar to how the interrupted flag is handled.